amanuensis/resources/filter_sets/__init__.py

```diff
@@ -0,0 +1,167 @@
+from amanuensis.resources.userdatamodel.search import get_filter_sets, update_filter_set
+import requests
+from amanuensis.settings import CONFIG_SEARCH_FOLDERS
+import os
+import glob
+from amanuensis.errors import NotFound, InternalError
+import json
+from cdislogging import get_logger
+from amanuensis.config import config
+
+logger = get_logger(__name__)
+
+
+def _load_data_files(file_name):
+    possible_configs = []
+
+    for folder in CONFIG_SEARCH_FOLDERS:
+        config_path = os.path.join(folder, file_name)
+        possible_files = glob.glob(config_path)
+        possible_configs.extend(possible_files)
+
+    if len(possible_configs) == 1:
+        file = possible_configs[0]
+    else:
+        raise NotFound(
+            "Could not find {}. Searched in the following locations: "
+            "{}".format(file_name, str(CONFIG_SEARCH_FOLDERS))
+        )
+    
+    with open(file, 'r') as file:
+        data = json.load(file)
+
+    return data
+
+
+def _get_explorer_selectable_values(portal_config):
+    selectable_values = {}
+    if len(portal_config["explorerConfig"]) == 0:
+        raise InternalError("No sections found in portal_config.explorerConfig")
+    for explorer in portal_config["explorerConfig"]:
+        selectable_values_by_tab = set()
+        for tab in explorer['filters']['tabs']:
+            selectable_values_by_tab.update(tab["fields"])
+        selectable_values[explorer["label"]] = selectable_values_by_tab
+
+    return selectable_values
+
+def _check_portal_config(filter_set, selectable_values):
+    """
+    This function will take the selectable explorer values
+    and check a filter-sets selectable values to validate whether
+    the filter-sets values are valid
+    Only has to be true for one explorer section
+    """
+    selected_keys = filter_set.filter_object["value"].keys()
+    filter_name = filter_set.name
+
+    for section, selectable_keys in selectable_values.items():
+        selected_keys_copy = set(selected_keys)
+        for key in selected_keys:
+            if key not in selectable_keys:
+                logger.info(f"The filter {filter_name} which contains the element {key} is not a valid selectable value in the section {section} of the data portal's selectable values")
+                break
+            else:
+                selected_keys_copy.remove(key)
+
+        if len(selected_keys_copy) == 0:
+            logger.info(f"The filter {filter_name} is valid for the section {section} of the data portal's selectable values")
+            return True
+    
+    logger.info(f"The filter {filter_name} is not valid for any section of the data portal's selectable values")
+    return False
```

Comment:
This should be by tab, not overall. A filter is associated to a specific tab, so that is the tab that should check.